### PR TITLE
added some error logging for extractors:  Data, Json, Query, and Path

### PR DESCRIPTION
**Pull request by Dowwie**

### CODE REVIEW

File: src/types/path.rs [hunk 1]
```diff
@@ -162,7 +162,10 @@ where
     #[inline]
     fn from_request(req: &HttpRequest, _: &mut Payload<P>) -> Self::Future {
         de::Deserialize::deserialize(PathDeserializer::new(req.match_info()))
-            .map(|inner| Path { inner })
+            .map(|inner| {
+                log::error!("Failed to deserialize during Path extractor");
```
Comment (fafhrd91) [inline src/types/path.rs]: i think this is wrong position and this should be debug

File: src/data.rs [hunk 2]
```diff
@@ -96,6 +96,7 @@ impl<T: 'static, P> FromRequest<P> for Data<T> {
         if let Some(st) = req.app_config().extensions().get::<Data<T>>() {
             Ok(st.clone())
         } else {
+            log::error!("Failed to construct App-level Data extractor");
```
Comment (fafhrd91) [inline src/data.rs]: i think this should debug or trace

**Review by fafhrd91 — CHANGES_REQUESTED** [review]

File: src/types/json.rs [hunk 3]
```diff
@@ -183,6 +183,7 @@ where
             JsonBody::new(req, payload)
                 .limit(limit)
                 .map_err(move |e| {
+                    log::error!("Failed to deserialize Json from payload");
```
Comment (fafhrd91) [inline src/types/json.rs]: this should be debug

File: src/types/query.rs [hunk 4]
```diff
@@ -122,6 +122,9 @@ where
     fn from_request(req: &HttpRequest, _: &mut Payload<P>) -> Self::Future {
         serde_urlencoded::from_str::<T>(req.query_string())
             .map(|val| Ok(Query(val)))
-            .unwrap_or_else(|e| Err(e.into()))
+            .unwrap_or_else(|e| {
+                log::error!("Failed during Query extractor deserialization");
```
Comment (fafhrd91) [inline src/types/query.rs]: should be debug as well

File: src/types/json.rs (hunk 3 above)
Comment (Dowwie) [reply src/types/json.rs]: fixed.

File: src/types/path.rs (hunk 1 above)
Comment (Dowwie) [reply src/types/path.rs]: fixed

File: src/types/query.rs (hunk 4 above)
Comment (Dowwie) [reply src/types/query.rs]: fixed

File: src/types/path.rs [hunk 5]
```diff
@@ -163,7 +163,10 @@ where
     fn from_request(req: &HttpRequest, _: &mut Payload<P>) -> Self::Future {
         de::Deserialize::deserialize(PathDeserializer::new(req.match_info()))
             .map(|inner| Path { inner })
-            .map_err(ErrorNotFound)
+            .map_err(|| {
+                log::debug!("Failed to deserialize during Path extractor");
```
Comment (fafhrd91) [inline src/types/path.rs]: one more thing. i think we should include `path` to log

**Review by fafhrd91 — APPROVED** [review]

**Pull request merged**